iceoryx_hoofs/include/iceoryx_hoofs/internal/containers/uninitialized_array.hpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+#define IOX_HOOFS_CONTAINERS_UNINITIALIZED_ARRAY_HPP
+
+#include "iceoryx_hoofs/iceoryx_hoofs_types.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace containers
+{
+/// @brief struct used as policy parameter in UninitializedArray to wrap an array with all entries zeroed
+/// @tparam ElementType is the array type
+/// @tparam Capacity is the array size
+template <typename ElementType, uint64_t Capacity>
+struct ZeroedBuffer
+{
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : required by low level UninitializedArray building block and encapsulated in abstraction
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
+    using element_t = cxx::byte_t[sizeof(ElementType)];
+    // AXIVION Next Construct AutosarC++19_03-A18.1.1 : required by low level UninitializedArray building block and encapsulated in abstraction
+    alignas(ElementType) element_t value[Capacity]{};
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
+};
+
+/// @brief struct used as policy parameter in UninitializedArray to wrap an array with all entries set to one
+/// @tparam ElementType is the array type
+/// @tparam Capacity is the array size
+template <typename ElementType, uint64_t Capacity>
+struct OneBuffer
+{
+    constexpr OneBuffer() noexcept
+        : value{}
+    {
+        constexpr uint8_t ONES = 0b11111111;
+        for (uint64_t i{0U}; i < Capacity; ++i)
```

Comment:
@MatthiasKillat I would disagree here since this is in our use case most of set to one at time compile time and `memset` is only runtime. 

Since this construct causes so much discussion I would be fine to remove it again.

Reply:
@elfenpiff The memory is always set at runtime, as only then the addresses are known.

Whether it is possible to have some `constexpr` function to work on compile time arrays is another matter (and not trivial to my knowledge). But this is only for compile time magic, say meta programming.